### PR TITLE
Bump kind to v0.4.0

### PR DESCRIPTION
**Pull request by jcsirot**
In e2e tests
- bump kind to v0.4.0
- add kube 1.15.0 in list of clusters in e2e tests

### CODE REVIEW

File: docker.Makefile [hunk 1]
```diff
@@ -3,8 +3,8 @@ include common.mk
 BUILD_BASE_IMAGE = golang:1.12.6-alpine3.10
 TEST_BASE_IMAGE = golang:1.12.6
 RUN_BASE_IMAGE = alpine:3.10.0
-KUBERNETES_VERSION ?= 1.13.3
-KIND_TAG = v$(KUBERNETES_VERSION)-$(TAG)
+KUBERNETES_VERSION ?= 1.13.7
```
Comment (thaJeztah) [inline docker.Makefile]: Should the default be changed to Kube 1.14.x? (I see the CLI is already on 1.14; https://github.com/docker/cli/blob/83aa71c45c0f996319bfe83d697cc0514c767d90/vendor.conf#L94-L100)

Comment (jcsirot) [reply docker.Makefile]: That's indeed a good point. WDYT @simonferquel @silvin-lubecki

Comment (jcsirot) [reply docker.Makefile]: done

**Comment by jcsirot** [comment]
@thaJeztah Yes, here the "official" images built for kind 0.4.0 are used. We can use custom built images with a different version of kube but I think it should be done in another PR.

**Pull request merged**